smslibrary/src/main/java/com/eis/random/RandomPeerGenerator.java

```diff
@@ -0,0 +1,24 @@
+package com.eis.random;
+
+import com.eis.communication.Peer;
+
+/**
+ * @param <P> the type of generated Peer.
+ * @author Riccardo De Zen.
+ */
+public interface RandomPeerGenerator<A, P extends Peer<A>> {
```

Comment:
Will do, I asked exactly because I couldn't decide, altough this means the library has to be imported as a test library too (I'm 99% sure of this, gotta check)

Reply:
It's not a big deal since it's probably not compiled when the app is built, but I'd have to check how Proguard works to confirm it